virttest/utils_net.py

```diff
@@ -3062,6 +3062,88 @@ def get_correspond_ip(remote_ip):
     return local_ip
 
 
+def get_linux_mac(session, nic):
+    """
+    Get MAC address by nic name
+    """
+    sys_path = "%s/%s" % (SYSFS_NET_PATH, nic)
+    pattern = "(\w{2}:\w{2}:\w{2}:\w{2}\:\w{2}:\w{2})"
+    if session.cmd_status("test -d %s" % sys_path) == 0:
+        mac_index = 1
+        show_mac_cmd = "cat %s/address" % sys_path
+        out = session.cmd_output(show_mac_cmd)
+    else:
+        pattern = "(ether|HWaddr) %s" % pattern
+        mac_index = 2
+        show_mac_cmd = "ifconfig %s || ip link show %s" % (nic, nic)
+        out = session.cmd_output(show_mac_cmd)
+    try:
+        return re.search(pattern, out, re.M | re.I).group(mac_index)
+    except Exception:
+        logging.error("No HWaddr/ether found for nic %s: %s" % (nic, out))
```

Comment:
I think no need raise error here.
if no match mac, it will return none.
and Its' caller don't mind the none return value